examples_tests/17.SimpleBulletIntegration/Camera.hpp

```diff
@@ -0,0 +1,320 @@
+		concatMatrix = core::matrix4SIMD::concatenateBFollowedByAPrecisely(projMatrix, core::matrix4SIMD(viewMatrix));
+	}
+
+	inline bool getLeftHanded() const { return leftHanded; }
+	
+	inline float getNearValue() const { return zNear; }
+
+	inline float getFarValue() const { return zFar; }
+
+	inline float getAspectRatio() const { return aspectRatio; }
+
+	inline float getFOV() const { return fovy; }
+
+	inline void setLeftHanded(bool _leftHanded = true)
+	{
+		leftHanded = _leftHanded;
+		recomputeViewMatrix();
+		recomputeProjectionMatrix();
+	}
+
+	inline void setNearValue(float zn)
+	{
+		zNear = zn;
+		recomputeProjectionMatrix();
+	}
+
+	inline void setFarValue(float zf)
+	{
+		zFar = zf;
+		recomputeProjectionMatrix();
+	}
+
+	inline void setAspectRatio(float aspect)
+	{
+		aspectRatio = aspect;
+		recomputeProjectionMatrix();
+	}
+
+	inline void setFOV(float fovy)
+	{
+		fovy = fovy;
+		recomputeProjectionMatrix();
+	}
+
+public:
+
+	void mouseProcess(const IMouseEventChannel::range_t& events)
+	{
+		for (auto eventIt=events.begin(); eventIt!=events.end(); eventIt++)
+		{
+			auto ev = *eventIt;
+
+			if(ev.type == ui::SMouseEvent::EET_CLICK && ev.clickEvent.mouseButton == ui::EMB_LEFT_BUTTON) {
+				if(ev.clickEvent.action == ui::SMouseEvent::SClickEvent::EA_PRESSED) {
+					mouseDown = true;
+				} else if (ev.clickEvent.action == ui::SMouseEvent::SClickEvent::EA_RELEASED) {
+					mouseDown = false;
+				}
+			}
+
+		}
+	}
+
+	void keyboardProcess(const IKeyboardEventChannel::range_t& events)
+	{
+		for (auto eventIt=events.begin(); eventIt!=events.end(); eventIt++)
+		{
+			auto ev = *eventIt;
+
+			if(ev.keyCode == ui::EKC_UP_ARROW || ev.keyCode == ui::EKC_W) {
+				if(ev.action == ui::SKeyboardEvent::ECA_PRESSED) {
+					keysDown[Keys::EKA_MOVE_FORWARD] = true;
+				} else if(ev.action == ui::SKeyboardEvent::ECA_RELEASED) {
+					keysDown[Keys::EKA_MOVE_FORWARD] = false;
+				}
+			}
+
+			if(ev.keyCode == ui::EKC_DOWN_ARROW || ev.keyCode == ui::EKC_S) {
+				if(ev.action == ui::SKeyboardEvent::ECA_PRESSED) {
+					keysDown[Keys::EKA_MOVE_BACKWARD] = true;
+				} else if(ev.action == ui::SKeyboardEvent::ECA_RELEASED) {
+					keysDown[Keys::EKA_MOVE_BACKWARD] = false;
+				}
+			}
+
+			if(ev.keyCode == ui::EKC_LEFT_ARROW || ev.keyCode == ui::EKC_A) {
+				if(ev.action == ui::SKeyboardEvent::ECA_PRESSED) {
+					keysDown[Keys::EKA_MOVE_LEFT] = true;
+				} else if(ev.action == ui::SKeyboardEvent::ECA_RELEASED) {
+					keysDown[Keys::EKA_MOVE_LEFT] = false;
+				}
+			}
+
+			if(ev.keyCode == ui::EKC_RIGHT_ARROW || ev.keyCode == ui::EKC_D) {
+				if(ev.action == ui::SKeyboardEvent::ECA_PRESSED) {
+					keysDown[Keys::EKA_MOVE_RIGHT] = true;
+				} else if(ev.action == ui::SKeyboardEvent::ECA_RELEASED) {
+					keysDown[Keys::EKA_MOVE_RIGHT] = false;
+				}
+			}
+		}
+	}
+
+	void update(double dt) {
+		float moveSpeed = 0.1f;
+		float rotateSpeed = 1.0f;
+
+		if(!mouseDown) {
+			return;
+		}
```

Comment:
you want to do this filtering of whether to move mouse or not in the `mouseProcess`

Reply:
basically if `mouseDown` then process a relative movement event, if not then skip